src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -976,4 +976,45 @@
       it { expect(response).to redirect_to(project_show_path(apache_maintenance_incident_project)) }
     end
   end
+
+  describe 'POST #move_path' do
+    context 'valid project' do
+      let(:repository) { create(:repository, project: apache_project) }
+      let(:path_element) { create(:path_element, repository: repository) }
+
+      context 'without direction' do
+        it 'raises an exception' do
+          expect { post :move_path, params: { project: apache_project } }.to raise_error ActionController::ParameterMissing
+        end
+      end
+
+      context 'direction up' do
+        before do
+          post :move_path, params: { project: apache_project, repository: repository.id, direction: 'up', path: path_element }
+        end
+
+        it 'goes up' do
+          expect(flash[:notice]).to eq("Path moved up successfully")
+          expect(response).to redirect_to({ action: :index, controller: :repositories, project: apache_project })
+        end
+      end
+
+      context 'direction down' do
+        before do
+          post :move_path, params: { project: apache_project, repository: repository.id, direction: 'down', path: path_element }
+        end
+
+        it 'goes down' do
+          expect(flash[:notice]).to eq("Path moved down successfully")
```

Comment:
I will also check that the position has change (you need 2 path elements for that).